src/dataflow/src/decode/avro.rs

```diff
@@ -72,23 +72,21 @@ impl DecoderState for AvroDecoderState {
     }
 
     /// give a session a key-value pair
-    fn give_key_value<'a>(
+    fn decode_upsert_value<'a>(
         &mut self,
-        key: Row,
         bytes: &[u8],
         coord: Option<i64>,
         upstream_time_millis: Option<i64>,
-        session: &mut PushSession<'a, (Row, Option<Row>, Timestamp)>,
-        time: Timestamp,
-    ) {
+    ) -> Result<Option<Row>, ()> {
         match block_on(self.decoder.decode(bytes, coord, upstream_time_millis)) {
             Ok(diff_pair) => {
                 self.events_success += 1;
-                session.give((key, diff_pair.after, time));
+                Ok(diff_pair.after)
             }
             Err(err) => {
                 self.events_error += 1;
-                error!("avro deserialization error: {}", err)
+                error!("avro deserialization error: {}", err);
+                Err(())
```

Comment:
Should we just return the error, rather than logging it? If there's a reason not to, we should probably comment (otherwise I'll come by in a month having forgotten and fix it). One maybe reason is that if these errors are fatal for the upsert stream, but 1. I think they shouldn't be and 2. we can always log it and discard higher up.

Reply:
I think the motivation is being able to log the error in a place where we still have enough context to know what kind of error it is so we can prefix the error with a description like "avro deserialization error". 

I could have the error be returned as a string just as in `decode_key`